### ISSUE TITLE
[package] ninja/1.9.0: Incompatible ABI for GCC 4.9

### ISSUE THREAD

**Issue by uilianries** [conan-io/conan-center-index#624]
When building installer packages (cmake, bison, ninja, meson ...) sometimes we have a problem related to the C library version which is incompatible to the present in the current Docker image. In the past, Bincrafters opted to use CentOS6 which solved 99% of cases, but now we using Ubuntu 16.04 (GCC5) which is failing when we need to build a package with GCC 4.9 because conanio/gcc49 is Ubuntu 14.04.

Suggestion, identify installer packages and use a old Linux distro when building, CentOS6 for instance

/cc @ericLemanissier 

### Package and Environment Details (include every applicable attribute)
  * Package Name/Version: **ninja/1.9.0**
  * Operating System+version: **Linux Ubuntu 18.04**
  * Compiler+version: **GCC 4.9**
  * Docker image: **conanio/gcc49**
  * Conan version: **conan 1.21.0**
  * Python version: **Python 3.8**

### Conan profile (output of `conan profile show default` or `conan profile show <profile>` if custom profile is in use)
```
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=gcc
compiler.libcxx=libstdc++
compiler.version=4.9
os=Linux
os_build=Linux
[options]
libdrm:shared=True
[build_requires]
[env]
```

### Steps to reproduce (Include if Applicable)
- Run Docker conanio/gcc49
- Build a project which requires Ninja as build_requires (e.g. [libdrm](https://github.com/bincrafters/conan-libdrm))

### Logs (Include/Attach if Applicable)
https://github.com/bincrafters/conan-libdrm/runs/379290528

**Comment by ericLemanissier** [comment]
this is the continuation of https://github.com/conan-io/conan-center-index/issues/213

**Comment by Croydon** [comment]
I guess the core question is:
How can we identify installer packages automatically?

**Comment by uilianries** [comment]
IIRC Conan center hook is able to do it, it checks if os_arch and build_arch are declared only

**Comment by uilianries** [comment]
That's it: https://github.com/conan-io/hooks/blob/master/hooks/conan-center.py#L131

**Comment by ericLemanissier** [comment]
Another take on this problem could that [this line](https://github.com/conan-io/conan-center-index/blob/master/recipes/ninja/all/conanfile.py#L61) is the cause of the bug, but it would only solve stdlib abi issue, not glibc abi issues.

**Comment by Croydon** [comment]
> That's it: https://github.com/conan-io/hooks/blob/master/hooks/conan-center.py#L131

The problem is that sometimes we do have more settings for the build process, which are only getting deleted in `package()`. So we can't know before a build for sure that it is an installer

**Comment by uilianries** [comment]
> > That's it: https://github.com/conan-io/hooks/blob/master/hooks/conan-center.py#L131 [](/conan-io/hooks/blob/HEAD@{2020-01-08T23:31:32Z}/hooks/conan-center.py#L131)
> 
> The problem is that sometimes we do have more settings for the build process, which are only getting deleted in `package()`. So we can't know before a build for sure that it is an installer

That's true, it's fragile, maybe we can check it based on package_id

**Comment by uilianries** [comment]
fixed by #2062
